source/Irrlicht/COpenGLDriver.h

```diff
@@ -676,6 +679,8 @@ namespace video
             //! Returns whether setting was a success or not.
             bool setActiveTexture(uint32_t stage, core::smart_refctd_ptr<IRenderableVirtualTexture>&& texture, const video::STextureSamplingParams &sampleParams);
 
+            void setActiveBindings(const COpenGLDescriptorSet* descriptorSets[4]);
+
             const GLuint& constructSamplerInCache(const uint64_t &hashVal);

```

Comment:
sampler objects are actually shareable, so nuke.

Reply:
@Crisspl introduce sampler objects (make descriptor set layout {immutable} and descriptor sets {combined image} take the object instead of SSamplerParams) and nuke this